File: tests/chaos/tools/leak_memory.py
_leaks = []

tool = {"description": "Tool that leaks memory", "input": {}}

async def execute(args, ctx):
    _leaks.append(bytearray(1024 * 1024))  # 1MB per call
    return {"leaked_buffers": len(_leaks), "total_mb": len(_leaks)}

File: tests/chaos/tools/leak_memory.rb
$leaks = []

tool description: "Tool that leaks memory",
     input: {}

execute do |args, ctx|
  $leaks << ("\x00" * 1024 * 1024)
  { "leaked_buffers" => $leaks.size, "total_mb" => $leaks.size }
end
